purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/PaywallTransition.kt

```diff
@@ -0,0 +1,68 @@
+package com.revenuecat.purchases.paywalls.components
+
+import com.revenuecat.purchases.InternalRevenueCatAPI
+import dev.drewhamilton.poko.Poko
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonClassDiscriminator
+
+@InternalRevenueCatAPI
+@Poko
+@Serializable
+class PaywallTransition(
+    @get:JvmSynthetic val type: TransitionType = TransitionType.Fade,
+    @get:JvmSynthetic
+    @SerialName("displacement_strategy")
+    val displacementStrategy: DisplacementStrategy,
+    @get:JvmSynthetic val animation: PaywallAnimation? = null,
+) {
+
+    /**
+     * Determines how the view being animated out is displaced by the view being animated in.
+     *
+     * A [GREEDY] displacement will result in the space being taken up by the incoming view
+     * *before* it attempts to transition into the view hierarchy.
+     *
+     * A [LAZY] displacement will not do this, instead it will result in shifting the layout
+     * as the new view inserts itself.
+     */
+    @InternalRevenueCatAPI
+    @Serializable
+    enum class DisplacementStrategy {
```

Comment:
For all enums and sealed classes, we should write a custom serializer that provides a default value in case there is an issue deserializing, you can look at sample of deserializers inheriting `EnumDeserializerWithDefault`.

The reason for this was that, if we add more elements to an enum or sealed class, this could cause older SDKs to stop parsing the response, causing the paywall to fail to render. So we decided that, unless it's critical for displaying the paywall, all enums and sealed classes should have defaults